calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -157,13 +157,18 @@ func (rh resourceHelper) GetObjectType() reflect.Type {
 // and if the resource already exists then we call the Update method.
 func (rh resourceHelper) Apply(client client.Interface, resource runtime.Object) (runtime.Object, error) {
```

Comment:
Given we can only do these actions on Resources should the resource parm actually just be of type ResourceObject